src/js/jsx/sections/layers/LayersPanel.jsx

```diff
@@ -105,13 +105,44 @@ define(function (require, exports, module) {
          */
         _lastScrolledTo: Immutable.List(),
 
+        /**
+         * Used to suppress scrolling into view when the selection is changed
+         * by clicking directly on a layer face.
+         *
+         * @private
+         * @type {boolean}
+         */
+        _suppressNextScrollTo: false,
+
         componentWillMount: function () {
             this._setTooltipThrottled = synchronization.throttle(os.setTooltip, os, 500);
             this._mountedLayerIDs = new Set();
         },
 
+        /**
+         * A capture-phase click handler for the container. Used to suppress
+         * scrolling into view when clicking on the layers panel directly.
+         *
+         * @private
+         * @param {SyntheticEvent} event
+         */
+        _handleContainerClickCapture: function (event) {
+            var containerNode = React.findDOMNode(this.refs.container);
+            if (event.target !== containerNode) {
+                this._suppressNextScrollTo = true;
+            }
+        },
+
         componentDidMount: function () {
             this._scrollToSelection(this.props.document.layers);
+
+            var containerNode = React.findDOMNode(this.refs.container);
+            containerNode.addEventListener("click", this._handleContainerClickCapture, true);
```

Comment:
The `LayerFace` click handler calls `event.stopPropagation()`, so this has to be attached at the capture phase instead of the bubbling phase. I'm not aware of a way to attach capture-phase handlers in JSX. Is there one?